modules/data.remote/R/call_remote_process.R

```diff
@@ -0,0 +1,400 @@
+##' @examples
+##' \dontrun{
+##' call_remote_process(settings)
+##' }
+##' @author Ayush Prasad
+##'
+
+call_remote_process <- function(settings){
+  
+  # information about the date variables used in call_remote_process -
+  # req_start, req_end : start, end dates requested by the user, the user does not be aware about the status of the requested file in the DB  
+  # start, end : effective start, end dates created after checking the DB status. These dates are sent to remote_process for downloading and processing data
+  # write_raw_start, write_raw_end : start, end dates which are used while inserting and updating the DB
+  # the "pro" version of these variables have the same meaning and are used to refer to the processed file
+
+  reticulate::import_from_path("remote_process", file.path("..", "inst"))
+  reticulate::source_python(file.path("..", "inst", "remote_process.py"))
+  
+  input_file <- NULL
+  stage_get_data <- NULL
+  stage_process_data <- NULL
+  raw_merge <- NULL
+  pro_merge <- NULL
+  existing_raw_file_path <- NULL
+  existing_pro_file_path <- NULL
+  
+  # extract the variables from the settings list
+  siteid <- settings$remotedata$siteid
+  sitename <- settings$remotedata$sitename
+  raw_mimetype <- settings$remotedata$raw_mimetype
+  raw_formatname <- settings$remotedata$raw_formatname
+  geofile <- settings$remotedata$geofile
+  outdir <- settings$remotedata$outdir
+  start <- settings$remotedata$start
+  end <- settings$remotedata$end
+  source <- settings$remotedata$source
+  collection <- settings$remotedata$collection
+  scale <- settings$remotedata$scale
+  if(!is.null(scale)){
+  scale <- as.double(settings$remotedata$scale)
+  scale <- format(scale, nsmall = 1)
+  }
+  projection <- settings$remotedata$projection
+  qc <- settings$remotedata$qc
+  if(!is.null(qc)){
+  qc <- as.double(settings$remotedata$qc)
+  qc <- format(qc, nsmall = 1)
+  }
+  algorithm <- settings$remotedata$algorithm
+  credfile <- settings$remotedata$credfile
+  pro_mimetype <- settings$remotedata$pro_mimetype
+  pro_formatname <- settings$remotedata$pro_formatname
+  out_get_data <- settings$remotedata$out_get_data
+  out_process_data <- settings$remotedata$out_process_data
+  
+  
+  dbcon <- PEcAn.DB::db.open(settings$database$bety)
+  flag <- 0
+
+  # collection dataframe used to map Google Earth Engine collection names to their PEcAn specific names
+  collection_lut <- data.frame(stringsAsFactors=FALSE,
+                          original_name = c("LANDSAT/LC08/C01/T1_SR", "COPERNICUS/S2_SR", "NASA_USDA/HSL/SMAP_soil_moisture"),
+                          pecan_code = c("l8", "s2", "smap")
+  )
+  getpecancode <- collection_lut$pecan_code
+  names(getpecancode) <- collection_lut$original_name
```

Comment:
Not sure why you have two variables (collection_lut and getpecancode) that store the identical information. Also, if you have registry files you could generate tables like this on the fly, rather than having a place developers have to update a hard-coded table every time they add a product.